File: teste.js
const { chromium } = require('playwright-core');
(async () => {
  let browser;
  try {
    browser = await chromium.launch();
  } catch (e) {
    console.log('LAUNCH_FAIL: ' + e.message.split('\n')[0]);
    process.exit(2);
  }
  const page = await browser.newPage();
  const erros = [];
  page.on('pageerror', e => erros.push('PAGEERROR: ' + e.message));
  page.on('console', m => { if (m.type() === 'error') erros.push('CONSOLE: ' + m.text()); });
  await page.goto('http://localhost:8099/index.html', { waitUntil: 'networkidle' }).catch(e => console.log('GOTO: '+e.message));

  // Preencher config
  await page.fill('#cfg-especialidade', 'Otorrinolaringologia');
  await page.fill('#cfg-horario', '08:00 – 12:00');
  await page.fill('#cfg-profissional', 'Dra. Maria Silva');

  // Adicionar 3 pacientes
  for (let i = 1; i <= 3; i++) {
    await page.fill('#pac-nome', `Paciente ${i}`);
    await page.fill('#pac-mae', `Mãe ${i}`);
    await page.fill('#pac-cpf', '12345678901');
    await page.fill('#pac-telefone', '11999998888');
    await page.fill('#pac-endereco', `Rua Teste, ${i}`);
    await page.click('#btn-submit-pac');
  }
  const contador = await page.textContent('#contador');
  console.log('Contador:', contador.trim());
  const cpfMask = await page.inputValue('#lista-corpo tr:first-child td:nth-child(6)').catch(()=>null);

  // Editar primeiro paciente
  await page.click('#lista-corpo tr:first-child button[data-act="edit"]');
  console.log('Titulo edicao:', (await page.textContent('#pac-form-title')).trim());
  await page.fill('#pac-nome', 'Paciente 1 Editado');
  await page.click('#btn-submit-pac');
  console.log('Nome apos edicao:', await page.textContent('#lista-corpo tr:first-child td:nth-child(3) .font-semibold').catch(()=>'?'));

  // Mover para baixo e reordenar
  await page.click('#lista-corpo tr:first-child button[data-act="down"]');
  console.log('Ordem apos down:', await page.$$eval('#lista-corpo tr td.font-semibold', els => els.map(e=>e.textContent)));

  // Remover ultimo
  page.on('dialog', d => d.accept());
  const trs = await page.$$('#lista-corpo tr');
  await trs[trs.length-1].$eval('button[data-act="del"]', b=>b.click());
  console.log('Contador apos remocao:', (await page.textContent('#contador')).trim());

  // Verificar CPF mascarado na tabela
  console.log('Celula CPF/SUS:', (await page.textContent('#lista-corpo tr:first-child td:nth-child(6)')).replace(/\s+/g,' ').trim());

  // Gerar PDF (intercepta download)
  const dlPromise = page.waitForEvent('download', { timeout: 30000 }).catch(()=>null);
  await page.click('#btn-pdf');
  const dl = await dlPromise;
  console.log('Download PDF:', dl ? dl.suggestedFilename() : 'FALHOU/NENHUM');
  if (dl) { const p = '/workspace/teste.pdf'; await dl.saveAs(p); console.log('PDF salvo:', p); }
  console.log('Status:', (await page.textContent('#pdf-status')).trim());

  console.log('ERROS JS:', erros.length ? erros.join(' | ') : 'nenhum');
  await browser.close();
})();

File: teste.py
from playwright.sync_api import sync_playwright

with sync_playwright() as pw:
    browser = pw.chromium.launch()
    page = browser.new_page(accept_downloads=True)
    erros = []
    page.on("pageerror", lambda e: erros.append(f"PAGEERROR: {e}"))
    page.on("console", lambda m: erros.append(f"CONSOLE-ERR: {m.text}") if m.type == "error" else None)
    page.goto("http://localhost:8099/index.html", wait_until="networkidle")

    # Configurações
    page.fill("#cfg-especialidade", "Otorrinolaringologia")
    page.fill("#cfg-horario", "08:00 – 12:00")
    page.fill("#cfg-profissional", "Dra. Maria Silva – UBS Centro")

    # Adicionar 3 pacientes
    for i in range(1, 4):
        page.fill("#pac-nome", f"Paciente {i}")
        page.fill("#pac-mae", f"Maria Souza {i}")
        page.fill("#pac-nascimento", "1990-05-12")
        page.fill("#pac-cpf", "12345678901")
        page.fill("#pac-sus", "712345678901234")
        page.fill("#pac-telefone", "11999998888")
        page.fill("#pac-endereco", f"Rua das Flores, {i} - Centro")
        page.click("#btn-submit-pac")
    print("Contador:", page.inner_text("#contador").strip())
    print("CPF mascarado:", page.inner_text("#lista-corpo tr:first-child td:nth-child(6)").split()[1] if True else "")

    # Editar primeiro
    page.click("#lista-corpo tr:first-child button[data-act='edit']")
    print("Modo edicao:", page.inner_text("#pac-form-title").strip())
    page.fill("#pac-nome", "Paciente 1 EDITADO")
    page.click("#btn-submit-pac")
    nomes = [t.strip() for t in page.locator("#lista-corpo tr td.pdf-x").evaluate_all("")] if False else \
            page.evaluate("[...document.querySelectorAll('#lista-corpo tr td:nth-child(3) .font-semibold')].map(e=>e.textContent)")
    print("Nomes apos edicao:", nomes)

    # Reordenar: descer o primeiro
    page.click("#lista-corpo tr:first-child button[data-act='down']")
    nomes2 = page.evaluate("[...document.querySelectorAll('#lista-corpo tr td:nth-child(3) .font-semibold')].map(e=>e.textContent)")
    print("Ordem apos descer:", nomes2)

    # Remover ultimo (aceita confirm)
    page.on("dialog", lambda d: d.accept())
    page.locator("#lista-corpo tr:last-child button[data-act='del']").click()
    page.wait_for_timeout(300)
    print("Contador apos remover:", page.inner_text("#contador").strip())

    # Verificar largura da tabela PDF vs pagina A4
    page.click("#btn-pdf")
    try:
        dl = page.expect_download(timeout=45000)
        with dl as info:
            pass
        pdf_path = "/workspace/teste.pdf"
        info.value.save_as(pdf_path)
        print("PDF gerado:", info.value.suggested_filename)
    except Exception as e:
        print("FALHA DOWNLOAD:", e)
    print("Status:", page.inner_text("#pdf-status").strip())

    # Medir dimensoes do conteudo PDF antes de limpar (re-renderizar sem salvar)
    dims = page.evaluate("""() => {
      const area = document.getElementById('pdf-area');
      return { widthOK: true };
    }""")
    print("Erros JS:", erros if erros else "nenhum")
    browser.close()
